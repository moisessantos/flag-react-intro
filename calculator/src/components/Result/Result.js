import React from 'react';
import PropTypes from 'prop-types';
import InputGroup from 'react-bootstrap/InputGroup'

const Result = ({ children, ...otherProps }) => (
  <InputGroup size="lg" {...otherProps}>
    {children || 0}
  </InputGroup>
);

Result.defaultProps = {
  children: '',
};

Result.propTypes = {
  children: PropTypes.string,
};

export default Result;
