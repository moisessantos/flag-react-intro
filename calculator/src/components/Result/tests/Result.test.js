import React from 'react';
import { shallow } from 'enzyme';
import Result from '..';

describe('Result tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<Result>500</Result>)
        expect(wrapper).toMatchSnapshot();
    });
    it('should render without children', () => {
        const wrapper = shallow(<Result></Result>)
        expect(wrapper).toMatchSnapshot();
    });
});