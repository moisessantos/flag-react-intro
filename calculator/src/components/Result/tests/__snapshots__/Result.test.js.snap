// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result tests should render as expected 1`] = `
<Bootstrap(InputGroup)
  size="lg"
>
  500
</Bootstrap(InputGroup)>
`;

exports[`Result tests should render without children 1`] = `
<Bootstrap(InputGroup)
  size="lg"
>
  0
</Bootstrap(InputGroup)>
`;
